Automattic/wp-calypso
# Taxonomies: Extract TermFormDialog and TermTreeSelector to blocks

This PR is part of this bigger PR #8670  that will allow editing and adding new taxonomies from different places in Calypso 

In this PR I Extracted `TermFormDialog` and `TermTreeSelector` to blocks This allows the two components to be reused on the Settings Page for editing taxonomies

I also extracted the "Add Category" button of the editor from the `TermFormDialog` Component

refs #8584

**Testing instructions**

There is no functional change in this PR, so just make sure we can add categories to a post from the Calypso Editor

cc @timmyc
